lisa/wlgen/rta.py

```diff
@@ -573,11 +584,16 @@ class Ramp(RTATask):
     :param cpus: the list of CPUs on which task can run.
                 .. note:: if not specified, it can run on all CPUs
     :type cpus: list(int)
+    
+    :param get_utils: when called with the index and load of a phase, 
+                      returns (util_min, util_max)
+    :type get_utils: callable
     """
 
     def __init__(self, start_pct=0, end_pct=100, delta_pct=10, time_s=1,
                  period_ms=100, delay_s=0, loops=1, sched_policy=None,
-                 priority=None, cpus=None):
+                 priority=None, cpus=None, 
+                 get_utils=lambda idx, load: (0, 1024)):
```

Comment:
Also nit: given that we don't know the count of phases when this gets called, does it make sense to keep the `idx` parameter?

I was originally thinking that we could even drop the callbacks and let the user specify the clamps manually (e.g. create the `RTATask` object, iterate over `.phases` and tweak accordingly - the good thing is you have the total count of phases here), but @douglas-raillard-arm made a point that the callback simplifies quite a few scenarios.

Also, seeing as you're duplicating this lambda in a few places, it could be worth having it defined as an actual method somewhere. The character count won't get better (it might get worse), but at least the default will live in a centralised place.